test/analysis/weightFunction.test.ts

```diff
@@ -71,7 +70,7 @@ describe('Weight Function correctly parses Argument Nodes if', () => {
             expect(getQueryTypeComplexity(queryAST, {}, typeWeights)).toBe(4);
         });
 
-        xtest('and the argument is passed in as a variable', () => {
+        test('and the argument is passed in as a variable', () => {
```

Comment:
I think we should add a test here for custom object weight of 0 and a custom scalar weight greater than 0 to make sure the weight function properly adds/multiplies.

Reply:
I've added some tests to check for this and added some more confirm the weight function is adding and multiplying complexities they we would expect with nesting and scalar lists. Check the expected outputs.

![Screen Shot 2022-07-07 at 11 45 47 AM](https://user-images.githubusercontent.com/89324687/177847580-4ccb9b89-8a88-4dcd-a082-3cf8bcf9e081.png)

Reply:
This is basically an integration test so it's nice the know these are passing needing the mockFunctions to be set properly.